### PR TITLE
Fix issue 199

### PR DESCRIPTION
**Pull request by ericmjonas**
Fix issue #199 and additional make `.done()` behave correctly (how did we never have a test for this?!)

### CODE REVIEW

File: pywren/future.py [hunk 1]
```diff
@@ -94,6 +102,10 @@ def result(self, timeout=None, check_only=False, throw_except=True, storage_hand
         if self._state == JobState.new:
             raise ValueError("job not yet invoked")
 
+        if check_only:
```
Comment (shivaram) [inline pywren/future.py]: why don't we write a new `check` function that we can call from within this function ? That way we can avoid that `FIXME` below ?

File: pywren/future.py [hunk 2]
```diff
@@ -172,36 +187,34 @@ def result(self, timeout=None, check_only=False, throw_except=True, storage_hand
...
+            if throw_except:
 
-        elif throw_except:
-
-            self._exception = call_invoker_result['result']
-            self._traceback = (call_invoker_result['exc_type'],
-                               call_invoker_result['exc_value'],
-                               call_invoker_result['exc_traceback'])
-
-            self._state = JobState.error
```
Comment (shivaram) [inline pywren/future.py]: We seem to have dropped this line `self._state = JobState.error` -- is this intentional ?

Comment (ericmjonas) [reply pywren/future.py]: we switched to using the proper `self._set-state(JobState.error)` in line 195

File: pywren/future.py (hunk 1 above)
Comment (ericmjonas) [reply pywren/future.py]: Yes that's the right thing to do but the three of us all have deadlines; writing `check()` creates _another_ "temporarily create the storage handler` point in the code and I don't want to think about that right now.  

basically our semantics are crap and we need a refactoring; right now I just want `done()` to work well enough that I can have a fast status check. 

Basically this is meant as a patch not a refactor :) 

I agree though this all stinks, it stinks to high heaven.

**Pull request merged**